### DISCUSSION TITLE
defaulting built-in apps to specific Families?

### DISCUSSION THREAD

**Discussion by godbout** [godbout/kindaVim.blahblah#48] [Ideas]
reported by kcmannem: https://github.com/godbout/kindaVim.theapp/issues/42

on one hand i like giving as little work as possible to the user. on the other hand sooner or later they'll have to deal with the `Families` Preferences. so, is it better to let the user discover early but not defaulting some settings, or is it better to default some settings and probably having the same issue but with other apps further down the line?

**Comment by jannis-baum** [comment]
I think both sides of this argument are very important! How about presenting a hint popup similar to the one you get when kV is sleeping?

Let's say the user opens app X for the first time since they installed kV and The Wizard knows X would be best off in the PG-R family. The user could get a popup stating this, along with a button to take them to the preference panel where they can make the adjustment.

This way they would get more of an *it just works* feeling while still becoming familiar with the concept of families.

**Reply by godbout** [reply]
yeah, i've added something in the menu bar so that kV tells you on which app you are, and you can send it directly to The Wizard. then The Wizard opens with the explanation/suggestions if there are any. it's better than before, but i agree, there's still work to do. it's gonna take some time to digest, and work out something i'm happy with. currently there's some engine work coming, and finally i'm gonna have to attack the `.` engine. probably getting something a bit easier for new comers will come after. it's definitely in my list.

**Reply by godbout** [reply]
just coming back to say i like your ideas (this one, and the other one). thanks :D great thinking. gonna be very useful and good to make kV better.

**Reply by jannis-baum** [reply]
Sounds great! Definitely looking forward to that `.`! I'm happy to help such a great project; at least with ideas if I can't open PRs🙈 I have been wanting to make something like kV and Wooshy for a while but just never found the time to start such a big project. So now I'm glad I found out you are already doing it and I get to enjoy haha

**Reply by godbout** [reply]
> at least with ideas if I can't open PRs

for the history, kV was open source for a while. but no OSS support. people had to be familiar with Xcode, and remove the license part etc to make it work. but even though i made it clear i wasn't doing support, still too many requests. also people weren't happy because i released almost one version a week, so they had a hard time keeping up to date 😂️ so ultimately i've closed it, now i ignore requests guilt-free and focus on subscribers and making kV better overall.

> I have been wanting to make something like kV and Wooshy for a while but just never found the time to start such a big project.

Wooshy is definitely less complex, but kV is a mastodon. you need a lot of info from an app in order to simulate Vim. there's more and more cross platform apps that respect no standard except their own, so it's a mega pain. many steps you can get wrong. plus well, Vim has a shitload of moves. that you then have to emulate for different apps/contexts. it's a lifetime project. coming up with the various engines took 1.5 years full-time (by full-time i mean 7 days a week 😂️). there's an easier way by far, which is using a Vim backend. kV started like this, but there's limitations that you reach quickly, like one for example is that screen moves don't work. so using a Vim backend is good for quick, small edits, but not if you want your Vim moves everywhere like i wanted (go down your list of emails, go down comboboxes, go up a webpage, etc). so it's basically writing Vim, but using macOS APIs, and dealing with buffers that you don't even own. but well i wanted it, so... 

also needs a massive test suite. too easy to make a change somewhere and break something. the last kV video was just this, the test suite running 😂️ (https://www.youtube.com/watch?v=p5dDG7YmOIA) half an hour. something like 6k+ tests by now. 

still a lot of things to do. and probably still some frictions for users. so please always reach out and we'll see how we can make kV better. thank you!

**Reply by jannis-baum** [reply]
🤯 well again, I'm really happy you're doing this! Hope you will never get tired of working on it haha

**Reply by godbout** [reply]
well as far as i'm concerned what i'm doing is not business software, it's self expression and pieces of art (so self aggrandizing hahahhaha). that means that as long as i use a computer, they live. as long as i live, they live. (already planned when/if i die, they go open source and free. but fyi i'm planning to live at least another 48 years.)
